### PR TITLE
Intelligent swap routes

### PR DESCRIPTION
**Pull request by sashimi36**
This is to introduce intelligent swap routes in the swap module and token price queries. It will now query price for every possible swap route to figure out the cheapest way for a swap.

### Steps to test
- In `useTokenSwap` hook, log the `tokenToTokenPrice` variable. The variable includes swap price plus selected swap route.
- Choose a pair for swap that has two possible routes; pass through swap & direct pool token swap;
- First, make sure there's enough liquidity in the direct pool for that given token pair; Then swap the tokens to validate if that exact pool was chosen for the swap;
- Then remove liquidity from the direct token pair pool and try to do the same swap;
- Observe if the pass through swap route was selected for the swap.

### CODE REVIEW

File: queries/tokenToTokenPriceQuery.ts [hunk 1]
```diff
@@ -10,62 +10,97 @@ import {
...
+
+  const prices: Array<TokenToTokenPriceQueryWithPoolsReturns> =
+    await Promise.all(pricingQueries)
+
+  /*
+   * pick the best price among all the available swap routes.
+   * the best price is the highest one.
+   * */
+  return prices.reduce((result, tokenPrice) => {
+    return result?.price < tokenPrice.price ? tokenPrice : result
```
Comment (ben2x4) [inline queries/tokenToTokenPriceQuery.ts:102]: Its a little confusing we call this variable price, price is a ratio ie 1 crab = 1.2 junox. What we have here is the output amount of the swap.

File: features/swap/hooks/useTokenSwap.tsx [hunk 2]
```diff
@@ -90,14 +96,22 @@ export const useTokenSwap = ({
         })
       }
 
+      // Smoke test
+      if (!passThroughPools?.length) {
```
Comment (ben2x4) [inline features/swap/hooks/useTokenSwap.tsx:100]: what is this?

File: package.json [hunk 3]
```diff
@@ -50,7 +50,7 @@
     "autoprefixer": "^10.3.1",
     "eslint": "7.30.0",
     "eslint-config-next": "11.0.1",
-    "next-bundle-junoblocks": "^1.0.0",
+    "next-bundle-junoblocks": "^1.0.1",
```
Comment (ben2x4) [inline package.json:53]: at this point its fine, but probable want to do update like this in a distict PR

**Review by ben2x4 — APPROVED** [review]
This looks good! Couple of small nits, up to you to fix or just merge as is.

File: queries/tokenToTokenPriceQuery.ts (hunk 1 above)
Comment (sashimi36) [reply queries/tokenToTokenPriceQuery.ts:102]: Agreed, could've been more clear here with the naming. Might be a good idea to address this in the follow up PR

**Pull request merged**